pinot-query-planner/src/main/java/org/apache/pinot/query/planner/physical/v2/opt/rules/LeafStageWorkerAssignmentRule.java

```diff
@@ -294,12 +295,14 @@ static TableScanWorkerAssignmentResult attemptPartitionedDistribution(String tab
     for (int partitionNum = 0; partitionNum < numPartitions; partitionNum++) {
       TablePartitionInfo.PartitionInfo info = partitionInfos[partitionNum];
       List<String> selectedSegments = new ArrayList<>();
-      if (info != null) {
+      List<String> segmentsForPartition = tablePartitionInfo.getSegmentsInPartition(partitionNum);
+      if (!segmentsForPartition.isEmpty()) {
```

Comment:
note: Though this check is not required, I found it makes the code slightly easier to read